### PR TITLE
docs: exec code block for shell commands in Test Nginx

### PR DESCRIPTION
**Pull request by bisakhmondal**
### Pre-submission checklist:

* [ ] Did you explain what problem does this PR solve? Or what new features have been added?
* [ ] Have you added corresponding test cases?
* [x] Have you modified the corresponding document?
* [x] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix/tree/master#community) first**

### CODE REVIEW

**Comment by membphis** [comment]
Have we used this feature in APISIX?

**Comment by bisakhmondal** [comment]
> Have we used this feature in APISIX?

Hi @membphis! Yes current all grpc tests using grpcurl are using this. 
PR https://github.com/apache/apisix/pull/5460/files
files: 1. https://github.com/apache/apisix/blob/master/t/node/grpc-proxy-unary.t
2. https://github.com/apache/apisix/blob/master/t/node/grpc-proxy-stream.t
3. https://github.com/apache/apisix/blob/master/t/node/grpc-proxy-mtls.t

**Review by membphis — APPROVED** [review]

**Comment by membphis** [comment]
cool job!

File: docs/en/latest/internal/testing-framework.md [hunk 1]
```diff
@@ -286,3 +286,24 @@ ONLY:
 --- response_body
 {"action":"get","count":0,"node":{"dir":true,"key":"/apisix/upstreams","nodes":{}}}
 ```
+
+### Executing Shell Commands
+
+It is possible to execute shell commands while writing tests in Test Nginx for apisix modules. We exeposes this feature via `exec` code block. The `stdout` of the executed process can be captured via `response_body` code block and `stderr` (if any) can be captured by filtering error.log through `grep_error_log`. Here is an example:
```
Comment (spacewander) [inline docs/en/latest/internal/testing-framework.md:292]:
```suggestion
It is possible to execute shell commands while writing tests in test-nginx for APISIX. We expose this feature via `exec` code block. The `stdout` of the executed process can be captured via `response_body` code block and `stderr` (if any) can be captured by filtering error.log through `grep_error_log`. Here is an example:
```

**Review by spacewander — APPROVED** [review]

**Pull request merged**